xrpl/utils/txn_parser/utils/order_book_parser.py

```diff
@@ -0,0 +1,217 @@
+)
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+LSF_SELL = 0x00020000
+
+
+def _get_offer_status(
+    node: NormalizedNode,
+) -> Literal["created", "partially-filled", "filled", "cancelled"]:
+    node_type = node["NodeType"]
+    if node_type == "CreatedNode":
+        return "created"
+    elif node_type == "ModifiedNode":
+        return "partially-filled"
+    else:  # node_type == "DeletedNode"
+        previous_fields = node.get("PreviousFields")
+        # a filled offer has previous fields
+        if previous_fields is not None:
+            return "filled"
+        # a cancelled offer has no previous fields
+        return "cancelled"
+
+
+def _derive_currency_amount(
+    currency_amount: Union[str, Dict[str, str]]
+) -> CurrencyAmount:
+    if isinstance(currency_amount, str):
+        return CurrencyAmount(currency="XRP", value=str(drops_to_xrp(currency_amount)))
+    else:
+        return CurrencyAmount(
+            currency=currency_amount["currency"],
+            issuer=currency_amount["issuer"],
+            value=currency_amount["value"],
+        )
+
+
+def _calculate_delta(
+    final_amount: CurrencyAmount,
+    previous_amount: CurrencyAmount,
+) -> str:
+    final_value = get_value(final_amount)
+    previous_value = get_value(previous_amount)
+    delta = final_value - previous_value
+    return str(delta)
+
+
+def _get_change_amount(
+    node: NormalizedNode,
+    status: Literal["created", "partially-filled", "filled", "cancelled"],
+    side: Literal["TakerGets", "TakerPays"],
+) -> Optional[CurrencyAmount]:
+    if status == "cancelled":
+        final_fields = node.get("FinalFields")
+        if final_fields is not None:
+            final_fields_amount = final_fields.get(side)
+            if final_fields_amount is not None:
+                return _derive_currency_amount(final_fields_amount)
+    if status == "created":
+        new_fields = node.get("NewFields")
+        if new_fields is not None:
+            new_fields_amount = new_fields.get(side)
+            if new_fields_amount is not None:
+                return _derive_currency_amount(new_fields_amount)
+    final_fields = node.get("FinalFields")
+    previous_fields = node.get("PreviousFields")
+    if final_fields is not None and previous_fields is not None:
+        final_fields_amount = final_fields.get(side)
+        previous_fields_amount = previous_fields.get(side)
+        if final_fields_amount is not None and previous_fields_amount is not None:
+            final_amount = _derive_currency_amount(final_fields_amount)
+            previous_amount = _derive_currency_amount(previous_fields_amount)
+            value = _calculate_delta(final_amount, previous_amount)
+            # final_amount is being reused because it makes
+            # it easier to return the changed amount.
+            final_amount["value"] = value
+            # From now on you could consider final_amount as changed_amount.
+            return final_amount
```

Comment:
Instead of this you can return a new object with just that field changed using the spread operator :) 

`return {**final_amount, "value": value}`

Source: https://how.wtf/spread-operator-in-python.html#object-literals

Reply:
Unfortunately your solution has typing problems because `final_amount` is a `TypedDict` (Argument 1 to "update" of "dict" has incompatible type "CurrencyAmount"; expected "Mapping[str, str]").
Alternatively I could do:
```python
change_amount = final_amount
change_amount["value"] = value
return change_amount
```

Reply:
I prefer that suggestion, although I'd recommend the name `changed_amount` instead of `change_amount`